cmd/tempo/app/modules.go

```diff
@@ -437,6 +437,12 @@ func (t *App) initCompactor() (services.Service, error) {
 }
 
 func (t *App) initStore() (services.Service, error) {
+	// Used by the local-blocs processor to flush RF1 blocks to storage.
+	// Only initialize if it's configured.
+	if t.cfg.Target == MetricsGenerator && t.cfg.StorageConfig.Trace.Backend == "" {
+		return services.NewIdleService(nil, nil), nil
+	}
```

Comment:
Other alternatives considered:
* New `MaybeStore` module that contains this logic
* Configurable feature flag, eg. `metrics_generator_flush_to_storage`

Reply:
Yes this is interesting. There is definitely a use case for generators without object store access.  I think the combination of both sounds the best?   The service has a dependency on the new module `MaybeStore/OptionalStore` which only starts if storage configured.  I like the approach of returning an IdleService.  Then the generator only flushes when `flush_to_storage=true`.  The generator can return a startup error if configured to flush but storage isn't.

Reply:
Changed! Now it uses both approaches.